knowledge_runtime/knowledge_runtime/services/duckdb_generator.py

```
@@ -0,0 +1,704 @@
+                # First row as headers
+                headers = [
+                    str(h) if h is not None else f"column_{i}"
+                    for i, h in enumerate(rows[0])
+                ]
+                data_rows = rows[1:]
+
+                # Convert to list of dicts for DataFrame
+                records = []
+                for row in data_rows:
+                    record = {}
+                    for i, (header, value) in enumerate(zip(headers, row)):
```

Ruff (0.15.12)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)